### ISSUE TITLE
Use local image if exist, provide required images list

### ISSUE THREAD

**Issue by yc-zc** [kubernetes-sigs/kubeadm-dind-cluster#278]
This is to solve [issue 277](https://github.com/kubernetes-sigs/kubeadm-dind-cluster/issues/277) and to further help running in a local machine without public network

**Comment by morningspace** [comment]
Hey @INT-ZC I have a PR #296 that has been merged which can exactly solve your problem... See also: #295 

BTW: I'm gonna create a video or post to describe how I use it... By using a shared local bridge network, I can have my 3-node k8s cluster be up & running in 1-2 mins :-)

**Comment by 15050050972** [comment]
I have configured a global vpn, and I configured docker proxy(https://docs.docker.com/config/daemon/systemd/#httphttps-proxy), I can wget www.google.com, and I can pull k8s.gcr.io/pause:3.1
```
[root@localhost dind]# docker pull k8s.gcr.io/pause:3.1
2019-04-02 01:57:01 INFO     connecting k8s.gcr.io:443 from 127.0.0.1:35436
2019-04-02 01:57:02 INFO     connecting k8s.gcr.io:443 from 127.0.0.1:35444
2019-04-02 01:57:02 INFO     connecting k8s.gcr.io:443 from 127.0.0.1:35458
3.1: Pulling from pause
2019-04-02 01:57:03 INFO     connecting k8s.gcr.io:443 from 127.0.0.1:35468
Digest: sha256:f78411e19d84a252e53bff71a4407a5686c46983a2c2eeed83929b888179acea
Status: Image is up to date for k8s.gcr.io/pause:3.1
```
```
[root@localhost dind]# wget www.google.com
--2019-04-02 02:01:35--  http://www.google.com/
Connecting to 127.0.0.1:8118... connected.
Proxy request sent, awaiting response... 2019-04-02 02:01:35 INFO     connecting www.google.com:80 from 127.0.0.1:37850
200 OK
Length: unspecified [text/html]
Saving to: ‘index.html’

    [ <=>                                                              ] 12,387      --.-K/s   in 0.008s  

2019-04-02 02:01:35 (1.43 MB/s) - ‘index.html’ saved [12387]

```
But why I still got an error when I try to execute `./dind-cluster-v1.13.sh up`?
thanks

**Comment by morningspace** [comment]
This can be closed since relative changes have been merged via #296. The basic idea is to grab required images at local and push to a private Docker registry, so that can make the cluster launch dramatically fast... See my post: [Launch multi-node Kubernetes cluster locally in one minute, and more…](https://morningspace.github.io/tech/k8s-run/)
